### PR TITLE
Apply clang-tidy suggestions to MsgLogger and WarnManager

### PR DESCRIPTION
**Pull request by lucafedeli88**
I started using the tool `clang-tidy` on WarpX (https://clang.llvm.org/extra/clang-tidy/).
In this PR I've applied some changes suggested by this tool to MsgLogger and WarnManager. The suggested changes
are supposed to  improve readability and to comply with (some of) the c++ core guidelines.

In this case I added few `static_cast<int>`, I set few methods as `static` and I used initializer lists for member variables.

### CODE REVIEW

File: Source/Utils/WarnManager.cpp [hunk 1]
```diff
@@ -187,7 +163,31 @@ std::string WarnManager::print_warn_msg(
...
+    std::stringstream ss;
+
+    ss << warn_header <<
+        std::string(line_size - static_cast<int>(warn_header.length()), '*') << "\n" ;
+
+    if(is_global){
+        ss << "* GLOBAL warning list  after " << " [ " <<  when << " ]\n*\n";
+    }
+    else{
+        auto const mpi_rank = amrex::ParallelDescriptor::MyProc();
```
Comment (ax3l) [inline Source/Utils/WarnManager.cpp:187]: CI note that no-MPI builds fail here :)

**Comment by ax3l** [comment]
Good call, thanks!

In case you like to run a few more on this section of the code, here are the things we enforce on openPMD-api:
https://github.com/openPMD/openPMD-api/blob/dev/.clang-tidy

File: Source/Utils/WarnManager.cpp (hunk 1 above)
Comment (lucafedeli88) [reply Source/Utils/WarnManager.cpp:187]: uh oh! Thanks... should work now...

**Comment by lucafedeli88** [comment]
> Good call, thanks!
> 
> In case you like to run a few more on this section of the code, here are the things we enforce on openPMD-api: https://github.com/openPMD/openPMD-api/blob/dev/.clang-tidy

For these specific files the only thing that comes out with your settings is
```
/home/luca/Projects/WarpX/Source/Utils/WarnManager.cpp:198:9: warning: 2 adjacent parameters of 'msg_formatter' of similar type ('const int') are easily swapped by mistake [bugprone-easily-swappable-parameters]
        const int line_size,
        ^~~~~~~~~~~~~~~~~~~~
/home/luca/Projects/WarpX/Source/Utils/WarnManager.cpp:198:19: note: the first parameter in the range is 'line_size'
        const int line_size,
                  ^~~~~~~~~
/home/luca/Projects/WarpX/Source/Utils/WarnManager.cpp:199:19: note: the last parameter in the range is 'tab_size'
        const int tab_size)

```
however, I am not sure that I want to change it....

Thanks anyway for the suggested checks! I would like to run them on few other files...

**Review by ax3l — APPROVED** [review]
Cool! :)

**Pull request merged**